packages/hub/src/components/layout/RootLayout/useForceChooseUsername.ts

```diff
@@ -0,0 +1,17 @@
+import { useSession } from "next-auth/react";
+
+import { chooseUsernameRoute } from "@/routes";
+
+export function useForceChooseUsername() {
+  const { data: session } = useSession();
+
+  if (
+    session?.user &&
+    !session?.user.username &&
+    !window.location.href.includes(chooseUsernameRoute())
+  ) {
+    // Next's redirect() is broken for components included from the root layout
```

Comment:
That seems incredibly annoying. Thanks for commenting though.

Reply:
This is an old comment, refactored to the new file.
Maybe they even fixed it, but I didn't bother to check this time.